### PR TITLE
chore: rid old StreamSink

### LINKED ISSUE #2937: Migrate StreamingSink to new sink variant instead of using compat
Blocked on #2934 

See [Architecture RFC](https://github.com/timberio/vector/blob/master/rfcs/2020-06-18-2625-architecture-revisit.md).

### PR DESCRIPTION
**Pull request by fanatid**
Closes #2937.

Currently, tcp/udp/unix flush each event, this is simple, but should seriously affect performance. I'll remove draft status once change data sending for better performance.

### CODE REVIEW

File: src/sinks/util/udp.rs [hunk 1]
```diff
@@ -83,103 +92,96 @@ impl UdpSink {
...
+                    bytes = &field::display(bytes.len())
                 );
-                match socket.send(&line) {
-                    Err(error) => {
-                        self.state = State::Backoff(self.next_delay01());
-                        error!(message = "send failed", %error);
-                        Ok(AsyncSink::NotReady(line))
+                while !bytes.is_empty() {
+                    match socket.send(&bytes).await {
+                        Ok(cnt) => bytes.advance(cnt),
```
Comment (juchiast) [inline src/sinks/util/udp.rs:163]: UDP does not guarantee the order of packets, so we cannot split one message into multiple packets, I added a warning for this in #4034.

**Comment by binarylogic** [comment]
@fanatid what's the status here?

**Comment by fanatid** [comment]
@binarylogic I need to add an internal buffer for proper acking. Planning to work on it on the current sprint.

**Comment by fanatid** [comment]
Closed in favour of #4597

**Pull request closed without merging**